include/filewriter/filewriter.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file filewriter.hpp
+ * @author Jeffin Sam
+ */
+#ifndef MLBOARD_FILE_WRITER_HPP
+#define MLBOARD_FILE_WRITER_HPP
+
+#include <mlboard/core.hpp>
+#include "sharedqueue.hpp"
+#include <proto/event.pb.h>
+#include "crc.hpp"
+
+namespace mlboard {
+
+/**
+ * Class resposnible for writing the event to a event file. The writing is a
+ * async operation which is running a separte thread. Look writeSummary()
+ * for more information.
+ */
+class FileWriter
+{
+ public:
+ /**
+  * Constructor Responsible for filewriter object.
+  *
+  * @param logdir Path to store the log files.
+  * @param maxQeueSize The maximum number of event to be store at a time.
+  * @param flushmilis Interval to perform the writing operation (milliseconds).
+  */
+  FileWriter(std::string logdir,
+            int maxQueueSize = 10,
+            size_t flushmilis = 5000);
+
+  /**
+   * A function to write the event in queue to event files. This function
+   * is intialised as a thread to complete the function without hindering
+   * any other option.
+   */
+  void writeSummary();
```

Comment:
I would probably use the same style as for mlpack so `WriteSummary` instead of `writeSummary`.

Reply:
Done